test/broker_test/broker_test.go

```diff
@@ -261,6 +263,52 @@ var _ = test.DescribeTestsFor(test.TestCase{
 					})
 				})
 
+				Context("when request body contains protected labels", func() {
+					BeforeEach(func() {
+						postBrokerRequestWithLabels.SetLabels(Object{
+							TenantLabelKey: []string{"test-tenant"},
+						})
+					})
+
+					It("returns 400", func() {
+						ctx.SMWithOAuth.POST(web.ServiceBrokersURL).WithJSON(postBrokerRequestWithLabels).
+							Expect().
+							Status(http.StatusBadRequest).
+							JSON().Object().
+							Keys().Contains("error", "description")
+
+						assertInvocationCount(brokerServer.CatalogEndpointRequests, 0)
+					})
+
+					Context("when request body contains multiple label objects", func() {
+						It("returns 400", func() {
+							ctx.SMWithOAuth.POST(web.ServiceBrokersURL).
+								WithHeader("Content-Type", "application/json").
+								WithBytes([]byte(fmt.Sprintf(`{
+								"name":        "broker-with-labels",
+								"broker_url":  "%s",
+								"description": "desc",
+								"credentials": {
+									"basic": {
+										"username": "%s",
+										"password": "%s"
+									}
+								},
+								"labels": {},
+								"labels": {
+									"%s":["test-tenant"]
+								}	
+							}`, brokerWithLabelsServer.URL(), brokerWithLabelsServer.Username, brokerWithLabelsServer.Password, TenantLabelKey))).
+								Expect().
+								Status(http.StatusBadRequest).
+								JSON().Object().
+								Keys().Contains("error", "description")
```

Comment:
some more specific error check
how can we be sure the error is due to the duplicate labels property?